Pinpoint.Core/PluginEngine.cs

```diff
@@ -20,9 +20,22 @@ public async Task AddPlugin(IPlugin @new)
             @new.Restore();
 
             // Ensure plugin hasn't been added and that it was able to load
-            if (Plugins.Contains(@new) || !await @new.TryLoad())
+            if (Plugins.Contains(@new))
```

Comment:
```suggestion
            if (!Plugins.Contains(@new))
```

Reply:
I'm not sure that makes sense, that would be the opposite of the current behavior. The check returns early if the plugin has already been added, so negating it would mean never adding a plugin unless it is already added.

Reply:
We only want to Add+Load a new plugin if it isn't already in the `Plugins` list, no?

Reply:
I can see why this might not work. But maybe the solution would instead be to have 
```suggestion
            if (Plugins.Contains(@new))
            {
                return;
            }
            
            try
            {
                var pluginLoaded = await @new.TryLoad();

                if (!pluginLoaded)
                {
                    return;
                }
            }   
            catch
            {
                // Plugin threw an exception while loading, so don't add it.
                return;
            }
```

Reply:
Yes, that also looks good to me 👍

Reply:
I've updated the pull request with these changes :)